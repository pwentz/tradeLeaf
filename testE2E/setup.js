import React, { Component } from 'react';
const { View, StyleSheet } = require('react-native');

import { Provider } from 'react-redux';

import MockApi from './mocks/MockApi';
const { setupApp } = require('../setupApp');

__DEV__ = true;

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    canOpenURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    getInitialURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
  };
});

jest.mock('ScrollView', () => jest.genMockFromModule('ScrollView'));

export const renderer = require('react-test-renderer');

const AccountRequirements = require('../src/components/register/AccountRequirements').default;
const App = require('../src/components/app/App').default;
const Avatar = require('../src/components/common/Avatar').default;
const Card = require('../src/components/matchboard/Card').default;
const LoginForm = require('../src/components/login/LoginForm').default;
const RegisterForm = require('../src/components/register/RegisterForm').default;
const TabBar = require('../src/components/app/TabBar').default;
const ChatPreview = require('../src/components/chat/ChatPreview').default;

export const components = {
  LoginForm,
  RegisterForm,
  Card,
  AccountRequirements,
  Avatar,
  TabBar,
  ChatPreview,
};

class TestRouteObserver {}

export function setupTestApp() {
  const api = new MockApi();

  const shouldLog = false;

  const { actions, store } = setupApp(api, shouldLog);

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'stretch',
      backgroundColor: 'white',
    },
  });

  const testRouteObserver = new TestRouteObserver();

  class TradeLeafApp extends Component {
    render() {
      return (
        <View style={styles.container}>
          <Provider store={store}>
            <App screenProps={{ actions, testRouteObserver }} />
          </Provider>
        </View>
      );
    }
  }

  return {
    TradeLeafApp,
    api,
    actions,
    dispatch: store.dispatch,
    testRouteObserver,
  };
}
