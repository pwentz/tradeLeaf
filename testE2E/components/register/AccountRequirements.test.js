import React from 'react';

import { components, renderer } from '../../setup';
import {
  authedUser
} from '../../mocks/mockData';

const noop = () => {}

it('<AccountRequirements />', () => {
  const AccountRequirements = components.AccountRequirements;
  const component = renderer.create(
    <AccountRequirements
      upload={noop}
      inProgress={false}
      isPhotoUploaded={false}
      hasOffers={false}
      isLocationEnabled={false}
      uploadedPhoto={authedUser.photo}
      userFirstName={authedUser.firstName}
      onLocationRequirementPress={noop}
      onContinuePress={noop}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
})
