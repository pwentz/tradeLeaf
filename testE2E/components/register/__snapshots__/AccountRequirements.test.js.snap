// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountRequirements /> 1`] = `
<View>
  <View
    style={
      Object {
        "height": 66.7,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/full-logo.png",
        }
      }
      style={
        Object {
          "aspectRatio": 1.5,
          "height": 333.5,
          "resizeMode": "contain",
          "width": 375,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": 233.45,
      }
    }
  />
  <View>
    <View
      style={
        Object {
          "alignItems": "stretch",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingBottom": 20,
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "borderRadius": 75,
                "height": 150,
                "width": 150,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "position": "relative",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://someofferphoto.png",
                }
              }
              style={
                Array [
                  Object {
                    "borderColor": "#B8B8B8",
                    "borderWidth": 1,
                  },
                  Object {
                    "borderRadius": 75,
                    "height": 150,
                    "width": 150,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "left": 197.02276707530646,
                  "position": "absolute",
                  "top": 230.26467203682392,
                }
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#27496B",
                      "borderRadius": 80,
                      "overflow": "hidden",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#F5F5F5",
                          "fontSize": 27.272727272727273,
                        },
                        Object {
                          "margin": 4,
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                      ]
                    }
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 400.2,
        "zIndex": -1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Welcome, 
      Buddy
      !
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Follow these steps to get started:
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 25,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 80,
                "margin": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#DFBA00",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 80,
                  "margin": 1,
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#F5F5F5",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#DFBA00",
                    "fontSize": 22,
                  },
                  Object {
                    "margin": 4,
                  },
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginLeft": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#B8B8B8",
              }
            }
          >
            Allow location services
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 25,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 80,
              "margin": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#DFBA00",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 80,
                "margin": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#F5F5F5",
              },
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#DFBA00",
                  "fontSize": 22,
                },
                Object {
                  "margin": 4,
                },
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#B8B8B8",
            }
          }
        >
          Post an offer
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "marginTop": 100.05,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderRadius": 80,
              "margin": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#DFBA00",
            },
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "borderRadius": 80,
                  "margin": 1,
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#F5F5F5",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 18,
                    "marginBottom": 26.68,
                    "marginLeft": 206.25000000000003,
                    "marginRight": 206.25000000000003,
                    "marginTop": 26.68,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "#DFBA00",
                  },
                ]
              }
            >
              continue
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
